v2/apiserver/internal/core/logs.go

```diff
@@ -150,7 +155,13 @@ func (l *logsService) Stream(
 	// in order to stream logs.
 	if err =
 		l.projectAuthorize(ctx, event.ProjectID, RoleProjectUser); err != nil {
-		return nil, err
+		// The principal wasn't an authorized Project user, but we also allow log
+		// access for the creator of the event so that gateways can send logs
+		// upstream.
+		if err =
+			l.authorize(ctx, system.RoleEventCreator, event.Source); err != nil {
```

Comment:
Would it be prudent to add a couple finer-grained unit tests around this?  (Say, role is right but event source doesn't match (error), both match (no error))?

Reply:
The authorizers all have their own tests, so they are known to function correctly.

[This case](https://github.com/brigadecore/brigade/blob/v2/v2/apiserver/internal/lib/authz/role_assignments_test.go#L32-L41) covers correct role, incorrect source.